src/post/content.ts

```diff
@@ -207,6 +211,77 @@ export class ContentPreparer {
         return new Set(mentions);
     }
 
+    /**
+     * Update existing hyperlinks with data-profile attribute and rel attributes,
+     * or create new hyperlinks for mentions that aren't already wrapped in <a> tags
+     */
+    private updateMentions(content: string, mentions: Mention[]) {
+        let preparedContent = content;
+
+        for (const mention of mentions) {
+            const escapedAccountApId = mention.account.apId
+                .toString()
+                .replace(/[.*+?^=!:${}()|\[\]\/\\]/g, '\\$&');
+            const escapedAccountUrl = mention.account.url
+                .toString()
+                .replace(/[.*+?^=!:${}()|\[\]\/\\]/g, '\\$&');
+            const escapedHandle = mention.name.replace(
+                /[.*+?^=!:${}()|\[\]\/\\]/g,
+                '\\$&',
+            );
+
+            // Handling existing hyperlinks in content
+            const existingLinkRegex = new RegExp(
+                `<a[^>]*href=["'](?:${escapedAccountApId}|${escapedAccountUrl})["'][^>]*>.*?</a>`,
+                'g',
+            );
```

Comment:
There may be trailing slashes in the href:

- https://example.com/users/bob
- https://example.com/users/bob/

So may want to do something like 
```
const apId = mention.account.apId.toString().replace(/\/+$/, '');
const url = mention.account.url.toString().replace(/\/+$/, '');
```

before adding it to the regex?

Reply:
Thanks for this! Nice suggestion